server/tso/global_allocator.go

```diff
@@ -121,6 +123,14 @@ func (gta *GlobalTSOAllocator) GenerateTSO(count uint32) (pdpb.Timestamp, error)
 		return pdpb.Timestamp{}, err
 	}
 	maxTSO.Logical += int64(count)
+	maxTSO.Logical = gta.timestampOracle.differentiateLogical(maxTSO.Logical)
+	// If the maxTSO's logical part is bigger than maxLogical, just add a updateTimestampGuard
+	// to the physical time and empty the logical part. We just need to make sure it's bigger than
+	// all the other Local TSOs.
+	if maxTSO.GetLogical() > maxLogical {
+		maxTSO.Physical += updateTimestampGuard.Milliseconds()
+		maxTSO.Logical = 0
+	}
```

Comment:
we still need to sign the suffix here.

Reply:
The Global TSO's suffix will always be zero, so it's no need to do the differentiate here.

Reply:
Got it. what about leave a comment to explain it here?